src/gfx_layout.cpp

```diff
@@ -64,15 +63,15 @@ Font::Font(FontSize size, TextColour colour) :
 template <typename T>
 static inline void GetLayouter(Layouter::LineCacheItem &line, std::string_view str, FontState &state)
 {
-	free(line.buffer);
+	typename T::CharType *buff_begin;
+	line.buffer = Layouter::LineCacheItem::Buffer(buff_begin = new typename T::CharType[str.size() + 1],
+		[](void *p) { delete[] reinterpret_cast<T::CharType *>(p); });
```

Comment:
this feels like it's trying to do too much in a single statement

Reply:
Yes, it is. But it's kind of deliberate. 

Now it looks like `buff_begin` is 'stealing' the pointer, so it can use it later, but importantly it does not have ownership.
If I were to move the allocation to the line before, the ownership would be firmly in `buff_begin` and it would look like I just pass a 'view' to `Buffer` instead of the actual ownership.

Also, how would it look if I allocated memory in one line with `new` and passed it into `std::unique_ptr` in the next line? That would be very weird, right? That's exactly what would be happening otherwise, but due to type erasure it's not as clear cut as I would like it to be.